src/qd8-f32-qb4w-gemm/gen/qd8-f32-qb4w-gemm-4x8c8-minmax-avx2.c

```diff
@@ -0,0 +1,423 @@
+    const __m256 vfilter_output_scale01 = _mm256_insertf128_ps(
+        _mm256_castps128_ps256(vfilter_output_scale0), vfilter_output_scale1, 1);
+    vout0x01 = _mm256_add_ps(vout0x01, _mm256_mul_ps(_mm256_cvtepi32_ps(vacc0x01), vfilter_output_scale01));
+    vout1x01 = _mm256_add_ps(vout1x01, _mm256_mul_ps(_mm256_cvtepi32_ps(vacc1x01), vfilter_output_scale01));
+    vout2x01 = _mm256_add_ps(vout2x01, _mm256_mul_ps(_mm256_cvtepi32_ps(vacc2x01), vfilter_output_scale01));
+    vout3x01 = _mm256_add_ps(vout3x01, _mm256_mul_ps(_mm256_cvtepi32_ps(vacc3x01), vfilter_output_scale01));
+    const __m128 vfilter_output_scale2 = _mm_broadcast_ss(&((const float*) w)[2]);
+    const __m128 vfilter_output_scale3 = _mm_broadcast_ss(&((const float*) w)[3]);
+    const __m256 vfilter_output_scale23 = _mm256_insertf128_ps(
+        _mm256_castps128_ps256(vfilter_output_scale2), vfilter_output_scale3, 1);
+    vout0x23 = _mm256_add_ps(vout0x23, _mm256_mul_ps(_mm256_cvtepi32_ps(vacc0x23), vfilter_output_scale23));
+    vout1x23 = _mm256_add_ps(vout1x23, _mm256_mul_ps(_mm256_cvtepi32_ps(vacc1x23), vfilter_output_scale23));
+    vout2x23 = _mm256_add_ps(vout2x23, _mm256_mul_ps(_mm256_cvtepi32_ps(vacc2x23), vfilter_output_scale23));
+    vout3x23 = _mm256_add_ps(vout3x23, _mm256_mul_ps(_mm256_cvtepi32_ps(vacc3x23), vfilter_output_scale23));
+    const __m128 vfilter_output_scale4 = _mm_broadcast_ss(&((const float*) w)[4]);
+    const __m128 vfilter_output_scale5 = _mm_broadcast_ss(&((const float*) w)[5]);
+    const __m256 vfilter_output_scale45 = _mm256_insertf128_ps(
+        _mm256_castps128_ps256(vfilter_output_scale4), vfilter_output_scale5, 1);
+    vout0x45 = _mm256_add_ps(vout0x45, _mm256_mul_ps(_mm256_cvtepi32_ps(vacc0x45), vfilter_output_scale45));
+    vout1x45 = _mm256_add_ps(vout1x45, _mm256_mul_ps(_mm256_cvtepi32_ps(vacc1x45), vfilter_output_scale45));
+    vout2x45 = _mm256_add_ps(vout2x45, _mm256_mul_ps(_mm256_cvtepi32_ps(vacc2x45), vfilter_output_scale45));
+    vout3x45 = _mm256_add_ps(vout3x45, _mm256_mul_ps(_mm256_cvtepi32_ps(vacc3x45), vfilter_output_scale45));
+    const __m128 vfilter_output_scale6 = _mm_broadcast_ss(&((const float*) w)[6]);
+    const __m128 vfilter_output_scale7 = _mm_broadcast_ss(&((const float*) w)[7]);
+    const __m256 vfilter_output_scale67 = _mm256_insertf128_ps(
+        _mm256_castps128_ps256(vfilter_output_scale6), vfilter_output_scale7, 1);
+    vout0x67 = _mm256_add_ps(vout0x67, _mm256_mul_ps(_mm256_cvtepi32_ps(vacc0x67), vfilter_output_scale67));
+    vout1x67 = _mm256_add_ps(vout1x67, _mm256_mul_ps(_mm256_cvtepi32_ps(vacc1x67), vfilter_output_scale67));
+    vout2x67 = _mm256_add_ps(vout2x67, _mm256_mul_ps(_mm256_cvtepi32_ps(vacc2x67), vfilter_output_scale67));
+    vout3x67 = _mm256_add_ps(vout3x67, _mm256_mul_ps(_mm256_cvtepi32_ps(vacc3x67), vfilter_output_scale67));
+
+    w = (const float*) w + 8;
+    }
+
+    const __m256 vout0x0213 = _mm256_hadd_ps(vout0x01, vout0x23);
```

Comment:
we only have 15 registers, curious to see the effects of this outside the block loop vs. inside on perf for smaller groupsize.